.github/workflows/fork-ci.yaml

```diff
@@ -0,0 +1,107 @@
+name: Fork Docker CI
+
+on:
+  push:
+    branches:
+    - network-operator-*
+    tags:
+    - network-operator-*
+
+jobs:
+  determine_docker_registry_and_tag:
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY_INTERNAL: nvcr.io/nvstaging/mellanox
+      REGISTRY_PUBLIC: nvcr.io/nvidia/cloud-native
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        sparse-checkout: .
+    - if: github.ref_type == 'branch'
+      name: Determine docker registry and tag (when git branch)
+      run: |
+        echo DOCKER_REGISTRY=$REGISTRY_INTERNAL >> $GITHUB_ENV
+        echo DOCKER_TAG=$(git rev-parse --short HEAD) >> $GITHUB_ENV  # short git commit hash
+    - if: github.ref_type == 'tag'
+      name: Determine docker registry and tag (when git tag)
+      run: |
+        echo DOCKER_REGISTRY=$(echo ${{ github.ref_name }} | sed 's/network-operator-//' | grep -q '-' && echo $REGISTRY_INTERNAL || echo $REGISTRY_PUBLIC) >> $GITHUB_ENV
```

Comment:
could you please describe when `$REGISTRY_PUBLIC` will be used?

Reply:
Only for releases that are tagged with "clean" versions (such as `23.5.0`) - i.e. that don't contain a '-' character (such as in `24.1.0-beta.3` or  tags).